api/provider/aws/environment_delete.go

```diff
@@ -43,6 +43,7 @@ func (e *EnvironmentProvider) Delete(environmentID string) error {
 	if err != nil && !strings.Contains(err.Error(), "does not exist") {
 		return err
 	}
+	err = nil
```

Comment:
this makes things pretty confusing, and can be deleted with my suggestions below